dulwich/client.py

```diff
@@ -1833,7 +1833,7 @@ def default_urllib3_manager(   # noqa: C901
     if proxy_server is None:
         for proxyname in ("https_proxy", "http_proxy", "all_proxy"):
             proxy_server = os.environ.get(proxyname)
-            if proxy_server is not None:
+            if proxy_server is not None and proxy_server != "":
```

Comment:
perhaps just "if not proxy_server"?

Reply:
This condition is much clearer. I just negated it to break the loop if a non empty value is set and added it to the pull request.